### PR TITLE
148 change tim1 and systick + + use hal for systick

### PR DESCRIPTION
**Pull request by Tsjerfell**

### CODE REVIEW

File: Nouveau PIR/firmware/source/timer.c [hunk 1]
```diff
@@ -130,10 +130,10 @@ void TIM_Soft_Stop(void)
 	if (TIM_Soft_Counting)
 	{
 		TIM_Soft_Counting = 0;
-		TIM_Soft_StopTick = __HAL_TIM_GET_COUNTER(&htim1);
+		TIM_Soft_StopTick =  HAL_GetTick();
```
Comment (hmathieu-insat) [inline Nouveau PIR/firmware/source/timer.c:133]: Are we sure that HAL_GetTick() returns the value of Systick->VAL?
Otherwise let's use the latter directly

**Review by hmathieu-insat — CHANGES_REQUESTED** [review]

File: Nouveau PIR/firmware/source/timer.c (hunk 1 above)
Comment (Tsjerfell) [reply Nouveau PIR/firmware/source/timer.c:133]: image
I think so

Comment (hmathieu-insat) [reply Nouveau PIR/firmware/source/timer.c:133]: fair enough

File: Nouveau PIR/firmware/source/usart_soft.c [hunk 2]
```diff
@@ -804,7 +804,7 @@ void USART_ProcessMessage(void)
 				communication_active = true;
 				communication_ready = true;
 
-				HAL_TIM_Base_Stop_IT(&htim1);
+				SYStick->CTRL |= ~SysTick_CTRL_ENABLE
```
[the remark this answers is not in the document]
Comment (Tsjerfell) [reply Nouveau PIR/firmware/source/usart_soft.c]: image
Il vaut 0 peut-être?

**Review by hmathieu-insat — APPROVED** [review]

**Pull request merged**